libs/libdecrypt/test/src/test.cpp

```diff
@@ -0,0 +1,22 @@
+#include "test.h"
+
+#ifndef PUBLIC_KEY_FILE
+#define PUBLIC_KEY_FILE "public_key.pem" // Default public key file
+#endif
+
+// Function to create a sample XML file for testing
+void createTestXMLFile(const std::string& filePath) {
```

Comment:
@Tulong4Dev Please check if the test provided is sufficient. If a local run on the unit test is o.k., I suggest to merge this PR. This PR is currently a blocking factor on another PR #2637